nilearn/glm/first_level/first_level.py

```diff
@@ -519,7 +519,8 @@ def fit(self, run_imgs, events=None, confounds=None,
         if self.verbose > 0:
             sys.stderr.write("\nComputation of %d runs done in %i seconds\n\n"
                              % (n_runs, time.time() - t0))
-
+        if design_matrices is not None and confounds is not None:
```

Comment:
I would expect it to be given at the beginning of the function rather than at the end, but otherwise the warning looks good.